### PR TITLE
ACMS-3623: Acquia cms telemetry changes as per acquia connector.

### PR DESCRIPTION
**Pull request by apathak18**
**Motivation**

Fixes ACMS-3623

**Proposed changes**
Making changes for telemetry as per acquia connector.

### CODE REVIEW

**Review by rajeshreeputra — CHANGES_REQUESTED** [review]

File: modules/acquia_cms_common/acquia_cms_common.services.yml [hunk 1]
```diff
@@ -45,14 +45,13 @@ services:
       - '@theme_handler'
       - '@string_translation'
       - '@extension.list.module'
+
   acquia_cms_common.telemetry:
     class: Drupal\acquia_cms_common\EventSubscriber\KernelTerminate\AcquiaCmsTelemetry
     arguments:
-      - '@extension.list.module'
+      - '@acquia_connector.telemetry_service'
```
[the remark this answers is not in the document]
Comment (apathak18) [reply modules/acquia_cms_common/acquia_cms_common.services.yml]: No DI is possible ~~ fixed

File: modules/acquia_cms_common/acquia_cms_common.info.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ type: module
 core_version_requirement: ~10.2.2
 dependencies:
   - acquia_purge:acquia_purge
+  - acquia_connector:acquia_connector
```
[the remark this answers is not in the document]
Comment (apathak18) [reply modules/acquia_cms_common/acquia_cms_common.info.yml]: Fixed

File: modules/acquia_cms_common/composer.json [hunk 3]
```diff
@@ -88,6 +88,9 @@
         "patches": {
             "drupal/core": {
                 "3301692 - Passing null to parameter to mb_strtolower() is deprecated with PHP 8.1": "https://git.drupalcode.org/project/drupal/-/merge_requests/2598.patch"
+            },
+            "drupal/acquia_connector": {
+                "ACMS-3417 - Remove amplitude and add sumologic for logging.": "https://gist.githubusercontent.com/rajeshreeputra/417e7f9cd9ba21c4a11292e0389f25a8/raw/6fc8e0e528ae5d97ecc9768cd42b872c879ba9f9/acquia_connector-sumologic.patch"
             }
```
[the remark this answers is not in the document]
Comment (apathak18) [reply modules/acquia_cms_common/composer.json:94]: How we get the latest changes of acquia connector and all the functionalities of the service

File: modules/acquia_cms_common/acquia_cms_common.services.yml [hunk 4]
```diff
@@ -45,14 +45,3 @@ services:
...
-    class: Drupal\acquia_cms_common\EventSubscriber\KernelTerminate\AcquiaCmsTelemetry
-    arguments:
-      - '@extension.list.module'
-      - '@config.factory'
-      - '@state'
-      - '%site.path%'
-      - '@datetime.time'
-      - '@logger.factory'
-    tags:
-      - { name: event_subscriber }
```
Comment (rajeshreeputra) [inline modules/acquia_cms_common/acquia_cms_common.services.yml:58]: Removal of acquia_cms_common.telemetry will break the BC.
cc @vishalkhode1 @chandan-singh7929

**Review by chandan-singh7929 — APPROVED** [review]
LGTM

File: modules/acquia_cms_common/acquia_cms_common.services.yml (hunk 4 above)
Comment (chandan-singh7929) [reply modules/acquia_cms_common/acquia_cms_common.services.yml:58]: I don't see any issue with this.

File: modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaConnectorTelemetryOverride.php [hunk 5]
```diff
@@ -25,9 +26,53 @@ class AcquiaConnectorTelemetryOverride extends AcquiaTelemetry {
   public function onTerminateResponse(KernelEvent $event) {
     // Check if telemetry opted,
     // then only trigger AcquiaTelemetry::onTerminateResponse($event).
-    if ($this->state->get('acquia_connector.telemetry.opted', TRUE)) {
+    if (\Drupal::state()->get('acquia_connector.telemetry.opted', TRUE) && PHP_SAPI !== 'cli') {
```
Comment (rajeshreeputra) [inline modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaConnectorTelemetryOverride.php:29]: @apathak18 remove everything from this PR except this change.

Comment (rajeshreeputra) [reply modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaConnectorTelemetryOverride.php:29]: As discussed with @apathak18 will keep this PR.

**Comment by rajeshreeputra** [comment]
@apathak18 can we close this PR.

**Pull request closed without merging**